front/lib/api/files/upsert.ts

```diff
@@ -0,0 +1,345 @@
+import type {
+  ConversationType,
+  FileUseCase,
+  PlainTextContentType,
+  Result,
+} from "@dust-tt/types";
+import { Err, isSupportedPlainTextContentType, Ok } from "@dust-tt/types";
+import { Writable } from "stream";
+import { pipeline } from "stream/promises";
+import { v4 as uuidv4 } from "uuid";
+
+import { getConversation } from "@app/lib/api/assistant/conversation";
+import { isJITActionsEnabled } from "@app/lib/api/assistant/jit_actions";
+import {
+  createDataSourceWithoutProvider,
+  upsertDocument,
+  upsertTable,
+} from "@app/lib/api/data_sources";
+import type { Authenticator } from "@app/lib/auth";
+import type { DustError } from "@app/lib/error";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import type { FileResource } from "@app/lib/resources/file_resource";
+import { SpaceResource } from "@app/lib/resources/space_resource";
+
+class MemoryWritable extends Writable {
+  private chunks: string[];
+
+  constructor() {
+    super();
+    this.chunks = [];
+  }
+
+  _write(
+    chunk: any,
+    encoding: BufferEncoding,
+    callback: (error?: Error | null) => void
+  ) {
+    this.chunks.push(chunk.toString());
+    callback();
+  }
+
+  getContent() {
+    return this.chunks.join("");
+  }
+}
+
+const notSupportedError: PostprocessingFunction = async ({ file }) => {
```

Comment:
```suggestion
const notSupportedError: PostprocessingFunction = ({ file }) => {
```

Reply:
Typescript complains (doesn't match the function sig)